### PR TITLE
Do not try to set groupId  from cache for null device

### PR DESCRIPTION
**Pull request by ltrung**
**Issue #1568 :**

**Description of changes:**
Currently we requires to call list device API before calling choosing device API. Otherwise, the device cache is not populated and the error `Device cache is not populated. Please make sure to call list devices first` will be thrown. However, this does not make sense in the case of calling null device. 
Update the logic when choosing null device to just set groupId to ''  instead of looking the groupId from device cache.

**Testing**

1. Have you successfully run `npm run build:release` locally? Yes
2. How did you test these changes? Add unit tests to cover this case.
    1. In the meeting demo, open console to delete the device cache (`delete app.audioVideo.deviceController.deviceInfoCache`).
    2. Call `app.audioVideo.chooseAudioInputDevice(null)`
    3. Verify that there is no error message log.
3. Can these changes be tested using the demo application? If yes, which demo application can be used to test it? Yes, meeting demo.
4. Do you add, modify, or delete public API definitions? If yes, has that been reviewed and approved? N/A
5. Do you change the wire protocol, e.g. the request method? If yes, has that been reviewed and approved? N/A

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -20,6 +20,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - Clarify why not use default downlink policy with simulcast.
 - Corrected argument `isUnifiedPlan` in `withBandwidthRestriction` to `isFirefox`. Also marked as deprecated since we no longer use it.
 - Update data message limit in the API Overview guide.
+- Do not try to set groupId  from cache for null device.
```
Comment (richnew10) [inline CHANGELOG.md:23]: I suggest you steal the changelog entry from my PR. These entries are supposed to be useful for the customer, not a commit message.

File: test/devicecontroller/DefaultDeviceController.test.ts [hunk 2]
```diff
@@ -997,6 +997,58 @@ describe('DefaultDeviceController', () => {
...
+      expect(e.notCalled).to.be.true;
+      expect(i.called).to.be.true;
+      expect(w.notCalled).to.be.true;
+
+      e.restore();
+      i.restore();
+      w.restore();
+    });
+
+    it('Use groupdId from device cache to prevent inifite loop', async () => {
```
Comment (richnew10) [inline test/devicecontroller/DefaultDeviceController.test.ts:1016]: infinite

File: src/devicecontroller/DefaultDeviceController.ts [hunk 3]
```diff
@@ -1486,10 +1486,13 @@ export default class DefaultDeviceController
...
-        kind,
-        this.getMediaTrackSettings(newDevice.stream)?.deviceId || ''
-      );
+      newDevice.groupId =
+        device !== null
+          ? this.getGroupIdFromDeviceId(
+              kind,
+              this.getMediaTrackSettings(newDevice.stream)?.deviceId || ''
+            )
+          : '';
```
Comment (richnew10) [inline src/devicecontroller/DefaultDeviceController.ts:1495]: Just do `newDevice.groupId = '';` in the `null` section above, and then here you can do a simple conditional.

**Review by richnew10 — CHANGES_REQUESTED** [review]

File: src/devicecontroller/DefaultDeviceController.ts (hunk 3 above)
Comment (ltrung) [reply src/devicecontroller/DefaultDeviceController.ts:1495]: The null section above is only for `audio` right?

File: test/devicecontroller/DefaultDeviceController.test.ts (hunk 2 above)
Comment (ltrung) [reply test/devicecontroller/DefaultDeviceController.test.ts]: Fixed.

**Review by richnew10 — DISMISSED** [review]

**Review by nainkunal933 — APPROVED** [review]

**Pull request merged**